src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -139,30 +139,62 @@ public void readMessage(Entity entity, List<Message> messages, String user) {
     }
   }
   /** Stores the User in Datastore. */
- public void storeUser(User user) {
-  Entity userEntity = new Entity("User", user.getEmail());
-  userEntity.setProperty("email", user.getEmail());
-  userEntity.setProperty("aboutMe", user.getAboutMe());
-  datastore.put(userEntity);
- }
+  public void storeUser(User user) {
+    Entity userEntity = new Entity("User", user.getEmail());
+    userEntity.setProperty("email", user.getEmail());
+    userEntity.setProperty("aboutMe", user.getAboutMe());
+    datastore.put(userEntity);
+  }
 
  /**
   * Returns the User owned by the email address, or
   * null if no matching User was found.
   */
- public User getUser(String email) {
+  public User getUser(String email) {
 
-  Query query = new Query("User")
+   Query query = new Query("User")
     .setFilter(new Query.FilterPredicate("email", FilterOperator.EQUAL, email));
-  PreparedQuery results = datastore.prepare(query);
-  Entity userEntity = results.asSingleEntity();
-  if(userEntity == null) {
+   PreparedQuery results = datastore.prepare(query);
+   Entity userEntity = results.asSingleEntity();
+   if(userEntity == null) {
    return null;
-  }
+   }
+
+   String aboutMe = (String) userEntity.getProperty("aboutMe");
+   User user = new User(email, aboutMe);
+
+   return user;
+   }
+
+   public List<UserMarker> getMarkers() {
+      List<UserMarker> markers = new ArrayList<>();
+
+      Query query = new Query("UserMarker");
+      PreparedQuery results = datastore.prepare(query);
+
+      for (Entity entity : results.asIterable()) {
+         try {
+            double lat = (double) entity.getProperty("lat");
+            double lng = (double) entity.getProperty("lng");
+            String content = (String) entity.getProperty("content");
+
+            UserMarker marker = new UserMarker(lat, lng, content);
+            markers.add(marker);
+         } catch (Exception e) {
+            System.err.println("Error reading marker.");
+            System.err.println(entity.toString());
+            e.printStackTrace();
+         }
+      }
+      return markers;
+   }
 
-  String aboutMe = (String) userEntity.getProperty("aboutMe");
-  User user = new User(email, aboutMe);
 
-  return user;
- }
+   public void storeMarker(UserMarker marker) {
```

Comment:
Blank line added. I think blank line 192 is unnecessary. Is it true @jonathantanmy ?

Reply:
Line 192 wasn't added - it's white, which means that it's existing, so don't worry about this, Sandy.